src/main/scala/org/broadinstitute/dsde/firecloud/HttpClient.scala

```diff
@@ -57,7 +57,10 @@ class HttpClient (requestContext: RequestContext) extends Actor
         log.debug("Got response: " + response)
         context.parent ! RequestCompleteWithHeaders(response, response.headers:_*)
       case Failure(error) =>
-        log.error("External request failed", error)
+        // Per docs: "the detail message associated with `cause` is not automatically incorporated
+        // into this exception 's detail message", so add it explicitly
+        val message = "External request failed to " + externalRequest.uri + " (" + error.getMessage + ")"
+        log.error(message, new Exception(message, error))
```

Comment:
My thinking is that the `message` and `cause` should be fully captured in an `Exception`. Otherwise, it seems possible that at some point the exception may become separated from its message (e.g. by being passed somewhere other than the logger and thrown).